src/main/java/com/training/springpractice/models/Movie.java

```diff
@@ -0,0 +1,100 @@
+package com.training.springpractice.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(name = "movies")
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+public class Movie {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    @Size(min = 50, max = 100)
+    private String title;
+
+    @NotNull
+    @Size(min = 100, max = 200)
+    private String description;
+
+    @NotNull
+    private Integer year;
+
+    @NotNull
+    @Enumerated(EnumType.STRING)
+    private RateType rate;
+    private Boolean deleted = false;
+
+    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
```

Comment:
What are the different types of CascadeTypes and please let me know about the differences between them and why you decided to use CascadeType.PERSIST.